frontend/src/modules/Metadata_Forms/components/createMetadataFormModal.js

```diff
@@ -0,0 +1,383 @@
+        })
+      );
+      if (!response.errors) {
+        setEnvironmentOptions(
+          response.data.listValidEnvironments.nodes.map((e) => ({
+            ...e,
+            value: e.environmentUri,
+            label: e.label
+          }))
+        );
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    } finally {
+      setLoading(false);
+      stopLoader();
+    }
+  };
+
+  useEffect(() => {
+    setVisibilityOptions(
+      Object.entries(visibilityDict).map((elem) => {
+        return { name: elem[0], value: elem[1] };
+      })
+    );
+
+    if (client && open) {
+      fetchEnvironments().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+      fetchOrganizations().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+      fetchGroups().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+    }
+  }, [client, open, dispatch]);
+
+  async function submit(values, setStatus, setSubmitting, setErrors) {
+    try {
+      let homeEntity = '';
+      if (values.visibility === visibilityDict.Team) {
+        homeEntity = values.group;
+      }
+      if (values.visibility === visibilityDict.Environment) {
+        homeEntity = values.environment;
+      }
+      if (values.visibility === visibilityDict.Organization) {
+        homeEntity = values.organization;
+      }
+      const response = await client.mutate(
+        createMetadataForm({
+          name: values.name,
+          description: values.description,
+          visibility: values.visibility,
+          SamlGroupName: values.owner,
+          homeEntity: homeEntity
+        })
+      );
+      if (!response.errors) {
+        setStatus({ success: true });
+        setSubmitting(false);
+        onApply();
+      } else {
+        setStatus({ success: false });
+        setErrors({ submit: response.errors[0].message });
+        setSubmitting(false);
+      }
+    } catch (err) {
+      console.error(err);
+      setStatus({ success: false });
+      setErrors({ submit: err.message });
+      setSubmitting(false);
+      dispatch({ type: SET_ERROR, error: err.message });
+    }
+  }
+
+  if (loading) {
+    return null;
+  }
+
+  return (
+    <Dialog maxWidth="md" fullWidth onClose={onClose} open={open} {...other}>
+      <Box sx={{ p: 3 }}>
+        <Typography
+          align="center"
+          color="textPrimary"
+          gutterBottom
+          variant="h4"
+        >
+          Create Metadata Form
+        </Typography>
+        <Formik
+          initialValues={{
+            name: '',
+            description: '',
+            visibility: visibilityDict.Global
+          }}
+          onSubmit={async (values, { setErrors, setStatus, setSubmitting }) => {
+            await submit(values, setStatus, setSubmitting, setErrors);
+          }}
```

Comment:
can we provide some `validationSchema` using Yup to ensure that:
- name is not null
- `owner` has an initial value and is not null
- description has max char length
- visibility is required

Also can we change name of `owner` to `SamlAdminGroupName` or similar - typically we use the name "owner" to represent the user's username/email that created the data.all resource rather than the group that manages the resource

Currently with testing I can create a form with no name and no group owner without any errors

Reply:
[8da0d13](https://github.com/data-dot-all/dataall/pull/1444/commits/8da0d130fd6cbb51d3fa161ec85627a191fd4334)